Tasks/VsTestV2/nondistributedtest.ts

```diff
@@ -0,0 +1,162 @@
+import * as xml2js from 'xml2js';
+import * as perf from 'performance-now';
+import * as process from 'process';
+import { InputDataContract } from './inputdatacontract';
+
+const runSettingsExt = '.runsettings';
+const testSettingsExt = '.testsettings';
+const sourceFilter = tl.getDelimitedInput('testAssemblyVer2', '\n', true);
+
+const inputDataContract: InputDataContract = undefined;
+let testAssemblyFiles = undefined;
+
+export function runNonDistributedTest() {
+    try {
+
+        console.log(tl.loc('runTestsLocally', 'vstest.console.exe'));
+        console.log('========================================================');
+
+        testAssemblyFiles = getTestAssemblies();
+        if (!testAssemblyFiles || testAssemblyFiles.length === 0) {
+            console.log('##vso[task.logissue type=warning;code=002004;]');
+            tl.warning(tl.loc('NoMatchingTestAssemblies', sourceFilter));
+            return;
+        }
+
+        const disableTIA = tl.getVariable('DisableTestImpactAnalysis');
+        if (disableTIA !== undefined && disableTIA.toLowerCase() === 'true') {
+            tl.debug('Disabling tia.');
+            inputDataContract.ExecutionSettings.TiaSettings.Enabled = false;
+        }
+
+        startDtaExecutionHost().then((code: number) => {
+            if (code !== 0) {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('VstestFailed'));
+                return;
+            }
+        });
+
+    } catch (err) {
+        tl.error(err);
+        tl.setResult(tl.TaskResult.Failed, tl.loc('VstestFailedReturnCode'));
+    }
+}
+
+async function startDtaExecutionHost() {
+    let dtaExecutionHostTool = tl.tool(path.join(inputDataContract.VsTestConsolePath, 'vstest.console.exe'));
+
+    inputDataContract.TestSelectionSettings.TestSourcesFile = createTestSourcesFile();
+    tl.cd(inputDataContract.TfsSpecificSettings.WorkFolder);
+    let envVars: { [key: string]: string; } = process.env;
+    dtaExecutionHostTool = tl.tool(path.join(__dirname, 'Modules/DTAExecutionHost.exe'));
+
+    // Invoke DtaExecutionHost with the input json file
+    const inputFilePath = utils.Helper.GenerateTempFile('input_' + uuid.v1() + '.json');
+    utils.Helper.removeEmptyNodes(inputDataContract);
+
+    try {
+        fs.writeFileSync(inputFilePath, JSON.stringify(inputDataContract));
+    } catch (e) {
+        tl.setResult(tl.TaskResult.Failed, `Failed to write to the input json file ${inputFilePath} with error ${e}`);
+    }
+
+    if (utils.Helper.isDebugEnabled()) {
+        utils.Helper.uploadFile(inputFilePath);
+    }
+
+    dtaExecutionHostTool.arg(['--inputFile', inputFilePath]);
+
+    utils.Helper.addToProcessEnvVars(envVars, 'DTA.AccessToken', tl.getEndpointAuthorization('SystemVssConnection', true).parameters.AccessToken);
+
+    // hydra: See which of these are required in C# layer. Do we want this for telemetry??
+    // utils.Helper.addToProcessEnvVars(envVars, 'DTA.AgentVersion', tl.getVariable('AGENT.VERSION'));
```

Comment:
add a task to fire telemetry from single agent flow

Reply:
done